### PR TITLE
promote certificates api to beta

### PR DESCRIPTION
**Pull request by mikedanese**
Mostly posting to see what breaks but also this API is ready to be promoted.

```release-note
Promote certificates.k8s.io to beta and enable it by default. Users using the alpha certificates API should delete v1alpha1 CSRs from the API before upgrading and recreate them as v1beta1 CSR after upgrading.
```

@kubernetes/api-approvers @jcbsmpsn @pipejakob

### CODE REVIEW

File: cmd/kubelet/app/bootstrap.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/golang/glog"
 
 	"k8s.io/apimachinery/pkg/types"
-	unversionedcertificates "k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/certificates/v1alpha1"
+	unversionedcertificates "k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/certificates/v1beta1"
```
Comment (liggitt) [inline cmd/kubelet/app/bootstrap.go]: should rename the import from `unversionedcertificates` :)

**Comment by liggitt** [comment]
doesn't removing the alpha type mean that any existing server with data in etcd will choke on upgrade?

**Comment by mikedanese** [comment]
@k8s-bot bazel test this

**Comment by mikedanese** [comment]
@liggitt I believe that we don't support upgrading clusters with alpha features enabled. The process for upgrading an alpha cluster should be: stop using alpha features (e.g. delete all alpha resources) and upgrade. We have done this before (e.g. we did this with deployment) so there's precedence and the policy is documented.

**Comment by cjcullen** [comment]
Changes lgtm. Would like a second approval on this being good to go.

**Comment by smarterclayton** [comment]
API changes are LGTM for alpha to beta.  However, can you ensure this plan is captured in the development docs so we can reference it next time?  I'm fine with a PR / issue separate from this.

**Comment by liggitt** [comment]
Also note in the release note that existing v1alpha1 CSR objects must be deleted before upgrading

**Comment by mikedanese** [comment]
@k8s-bot verification test this
@k8s-bot unit test this
@k8s-bot gke smoke test this

**Comment by spxtr** [comment]
@k8s-bot test this
https://github.com/kubernetes/test-infra/issues/1598

**Comment by mikedanese** [comment]
@k8s-bot unit test this

**Comment by mikedanese** [comment]
@k8s-bot cross build this

**Pull request merged**